llvm/lib/Target/AMDGPU/SIISelLowering.cpp

```diff
@@ -7280,6 +7280,81 @@ static SDValue lowerLaneOp(const SITargetLowering &TLI, SDNode *N,
   return DAG.getBitcast(VT, UnrolledLaneOp);
 }
 
+static SDValue lowerWaveShuffle(const SITargetLowering &TLI, SDNode *N,
+                                    SelectionDAG &DAG) {
+  EVT VT = N->getValueType(0);
+  assert(VT.getSizeInBits() == 32);
```

Comment:
Nit: this shouldn't be an assert because there is nothing to stop the user from calling the intrinsic with a different type like i64. Instead you can `return SDValue()` here and the compiler should fail with a nicer "cannot select" diagnostic.